test/integration/apiserver/cel/validatingadmissionpolicy_test.go

```diff
@@ -0,0 +1,461 @@
+	if strings.Contains(err.Error(), "evaluation error") {
+		t.Fatalf("CEL rule evaluation failed: %s", err)
+	}
+}
+
+func TestValidation(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.CELValidatingAdmission, true)()
+	server, err := apiservertesting.StartTestServer(t, nil, nil, framework.SharedEtcd())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.TearDownFn()
+
+	config := server.ClientConfig
+
+	client, err := clientset.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	configMap := makeConfigParams("policy-params", map[string]string{
+		"namespaceSuffix": "-k8s",
+	})
+	if _, err := client.CoreV1().ConfigMaps("default").Create(context.TODO(), configMap, metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	paramKind := &admissionregistrationv1alpha1.ParamKind{
+		APIVersion: "v1",
+		Kind:       "ConfigMap",
+	}
+	policy := withParams(paramKind, withNamespaceMatch(withFailurePolicy(admissionregistrationv1alpha1.Fail, makePolicy("test-policy"))))
+	policy.Spec.Validations = []admissionregistrationv1alpha1.Validation{
+		{
+			Expression: "object.name.endsWith(params.data.namespaceSuffix)",
+		},
+	}
+	if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), policy, metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	binding := makeBinding("test-policy-binding", "test-policy", "policy-params")
+
+	if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicyBindings().Create(context.TODO(), binding, metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	// allow time for reconciliation
+	time.Sleep(time.Second)
+
+	namespace := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-namespace",
+		},
+	}
+
+	if _, err := client.CoreV1().Namespaces().Create(context.TODO(), namespace, metav1.CreateOptions{}); err != nil {
+		checkForFailedRule(t, err)
+	}
+}
+
+func TestNoResource(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.CELValidatingAdmission, true)()
+	server, err := apiservertesting.StartTestServer(t, nil, nil, framework.SharedEtcd())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.TearDownFn()
+
+	config := server.ClientConfig
+
+	client, err := clientset.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	policy := withParams(&admissionregistrationv1alpha1.ParamKind{
+		APIVersion: "v1",
+		Kind:       "ConfigMap",
+	},
+		withNamespaceMatch(withFailurePolicy(admissionregistrationv1alpha1.Fail, makePolicy("test-policy"))))
+	policy.Spec.Validations = []admissionregistrationv1alpha1.Validation{
+		{
+			Expression: "true",
+		},
+	}
+	if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), policy, metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	/*
```

Comment:
This test ideally creates a `ConfgMap` (temporarily resulting in a valid policy + policy binding) and then deletes it (invalidating the binding) - I believe that part of the controller is not implemented yet (so the code is commented out), so I'm leaving this comment here as a reminder (and in case others have further insight).